include/aws/common/array_list.inl

```diff
@@ -77,12 +77,14 @@ void aws_array_list_init_static(
 
 AWS_STATIC_IMPL
 bool aws_array_list_is_valid(const struct aws_array_list *AWS_RESTRICT list) {
+    if(!list) {
+        return false;
+    }
     size_t required_size;
     bool required_size_is_valid = aws_mul_size_checked(list->length, list->item_size, &required_size) == AWS_OP_SUCCESS;
     bool current_size_is_valid = list->current_size >= required_size;
-    bool item_size_is_valid = list->item_size != 0;
```

Comment:
Why is this being removed?

Reply:
I tested the invariants with all functions and it turns out this particular one is not an invariant for the `aws_array_list` structure, only for the `aws_array_list_back` function. I'll include the pre- and post-conditions in all functions on a separate PR.